tools/convert/main.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
```

Comment:
Does `tools/convert` need to be part of this PR or could it be added separately, either in parallel or after this PR?